src/app/ReadClient.cpp

```diff
@@ -205,40 +206,52 @@ CHIP_ERROR ReadClient::SendReadRequest(ReadPrepareParams & aReadPrepareParams)
 
     VerifyOrReturnError(ClientState::Idle == mState, err = CHIP_ERROR_INCORRECT_STATE);
 
+    Span<AttributePathParams> attributePaths(aReadPrepareParams.mpAttributePathParamsList,
+                                             aReadPrepareParams.mAttributePathParamsListSize);
+    Span<EventPathParams> eventPaths(aReadPrepareParams.mpEventPathParamsList, aReadPrepareParams.mEventPathParamsListSize);
+    Span<DataVersionFilter> dataVersionFilters(aReadPrepareParams.mpDataVersionFilterList,
+                                               aReadPrepareParams.mDataVersionFilterListSize);
+    for (auto & attribute : attributePaths)
+    {
+        attribute.mpReadClient = this;
```

Comment:
As described above, don't think we need this.

Reply:
for instance, first subscribe with C1, then read using same cache for C1, we need mpReadClient to avoid the removal of read request paths from first subscribe when second read closes. It may not be big deal, user may or may not suscribe/read same path, but in case, I hope we can safely handle this situation.

Reply:
removed ,thanks